src/lib/DefaultRuleManager.ts

```diff
@@ -62,7 +62,14 @@ export class DefaultRuleManager implements RuleManager {
 			const engineGroups = ruleGroups.filter(g => g.engine === e.getName());
 			const engineRules = rules.filter(r => r.engine === e.getName());
 			const engineTargets = await this.unpackTargets(e, targets);
-			ps.push(e.run(engineGroups, engineRules, engineTargets));
+			this.logger.trace(`For ${e.getName()}, found ${engineGroups.length} groups, ${engineRules.length} rules, ${engineTargets.length} targets`);
+			if (engineRules.length > 0 && engineTargets.length > 0) {
```

Comment:
We'll invoke execution of an engine only when it has both chosen rules and target files to scan.

Reply:
Will this also prevent PMD from running when it has no rules?

Reply:
I hope yes - though I haven't tested it yet.